lib/libstatcounters/statcounters.h

```diff
@@ -31,90 +31,45 @@
 #ifndef STATCOUNTERS_H
 #define STATCOUNTERS_H
 
-#ifndef __has_extension
-#define __has_extension(x) 0
-#endif
-#if __has_extension(attribute_deprecated_with_message)
-#define DEPRECATED(x) __attribute__((deprecated(x)))
-#else
-#define DEPRECATED(x) __attribute__((deprecated))
-#endif
-
-#include <stdio.h>
 #include <stdint.h>
+#include <stdio.h>
 
-// counters bank
-#define STATCOUNTERS_MAX_MOD_CNT 12
-typedef struct statcounters_bank
-{
-    uint64_t itlb_miss;
-    uint64_t dtlb_miss;
-    uint64_t cycle;
-    uint64_t inst;
-    uint64_t inst_user;
-    uint64_t inst_kernel;
-    uint64_t imprecise_setbounds;
-    uint64_t unrepresentable_caps;
-    uint64_t icache[STATCOUNTERS_MAX_MOD_CNT];
-    uint64_t dcache[STATCOUNTERS_MAX_MOD_CNT];
-    uint64_t l2cache[STATCOUNTERS_MAX_MOD_CNT];
-    uint64_t mipsmem[STATCOUNTERS_MAX_MOD_CNT];
-    uint64_t tagcache[STATCOUNTERS_MAX_MOD_CNT];
-    uint64_t l2cachemaster[STATCOUNTERS_MAX_MOD_CNT];
-    uint64_t tagcachemaster[STATCOUNTERS_MAX_MOD_CNT];
-} statcounters_bank_t;
+#if defined(__mips__)
```

Comment:
Should be using MD subdirectories added to the include search path with a statcounters_md.h